src/backend/distributed/utils/ruleutils_11.c

```diff
@@ -7264,7 +7264,7 @@ get_from_clause_item(Node *jtnode, Query *query, deparse_context *context)
 		if (j->alias != NULL)
 		{
 			appendStringInfo(buf, " %s",
-							 quote_identifier(j->alias->aliasname));
+							 quote_identifier(get_rtable_name(j->rtindex, context)));
```

Comment:
This file (and `ruleutils_10.c`) is mostly copy/pasted versions of PostgreSQL's `postgres/backend/utils/adt/ruleutils.c`, with the purpose of just making some of the functions we need public. Are there any solutions to this issue that doesn't involve changing `ruleutils` files?

Reply:
Alternative would be to rely on shadowing & have both selects resolve as `join_alias` as opposed to the inner resolving `join_alias_1`